File: yh_tutorial/yh_tutorial_1/scripts/yh_pub_1.py
#!/usr/bin/python
# -*- coding: utf8 -*-
import rospy
from yh_tutorial_1.msg import yh_msg_1

def talker():
    pub = rospy.Publisher("yh_topic_1", yh_msg_1, queue_size=100)
    rospy.init_node("yh_pub_1", anonymous=True)
    loop_rate = rospy.Rate(2)
    cnt = 0
    msg = yh_msg_1()
    while not rospy.is_shutdown():
        msg.stamp = rospy.Time.now()
        msg.data = cnt
        pub.publish(msg)
        cnt += 1
        loop_rate.sleep()

if __name__ == "__main__":
    try:
        talker()
    except rospy.ROSInterruptException:
        pass

File: yh_tutorial/yh_tutorial_1/src/yh_pub_1.cpp
#include "ros/ros.h"
#include "yh_tutorial_1/yh_msg_1.h"

int main(int argc, char ** argv)
{
    ros::init(argc, argv, "yh_pub_1");
    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<yh_tutorial_1::yh_msg_1>("yh_topic_1", 100);
    ros::Rate loop_rate(2);
    int cnt = 0;
    yh_tutorial_1::yh_msg_1 msg;
    while (ros::ok())
    {
        msg.stamp = ros::Time::now();
        msg.data = cnt;
        cnt++;
        pub.publish(msg);
        loop_rate.sleep();
    }

    return 0;
}